src/Core/System/Snippet/Command/Util/CountryAgnosticFileLinter.php

```diff
@@ -0,0 +1,215 @@
+        if ($specificCollection->count() === 0) {
+            return $lintedFileStruct;
+        }
+
+        $translationCollection = $lintedFileStruct->getCompleteCollection();
+        foreach ($specificCollection as $countrySpecificFile) {
+            // If no agnostic file exists, $countrySpecificFile is content for `fixFilenames` to be fixed
+            if ($translationCollection->get($countrySpecificFile->getAgnosticPath()) === null) {
+                $lintedFileStruct->addFixableFile($countrySpecificFile);
+            }
+        }
+
+        return $lintedFileStruct;
+    }
+
+    private function getFinder(LintedTranslationFileOptions $options): Finder
+    {
+        $finder = (new Finder())
+            ->files()
+            ->ignoreUnreadableDirs()
+            ->ignoreDotFiles(true)
+            ->ignoreVCS(true)
+            ->exclude([
+                'node_modules',
+                'vendor',
+                'bin',
+                'static',
+                // Translations of languages fetched from crowdin should not be linted
+                'SwagLanguagePack/src/Resources/snippet',
+                'SwagLanguagePack/src/Resources/app/administration/src/snippet',
+                ...$options->ignoredPaths,
+            ])
+            ->name([SnippetPatterns::CORE_SNIPPET_FILE_PATTERN, SnippetPatterns::ADMIN_SNIPPET_FILE_PATTERN])
+            ->sortByName(true);
+
+        if ($options->dir) {
+            $finder->in($options->dir);
+        } elseif (empty($options->extensionPaths)) {
+            $finder->in('src');
+
+            if ($options->isAll) {
+                $finder->in('custom');
+            }
+        } else {
+            $finder->in($this->getExtensionPaths($options));
+        }
+
+        return $finder;
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    private function getExtensionPaths(LintedTranslationFileOptions $options): array
+    {
+        $criteria = (new Criteria())->addFilter(new EqualsAnyFilter('name', $options->extensionPaths));
+        $context = Context::createCLIContext();
+
+        $plugins = $this->pluginRepository->search($criteria, $context)->getEntities();
+        $apps = $this->appRepository->search($criteria, $context)->getEntities();
+
+        $extensionPaths = [
+            ...$plugins->map(static fn (PluginEntity $plugin) => $plugin->getPath()),
+            ...$apps->map(static fn (AppEntity $app) => $app->getPath()),
+        ];
+
+        if (empty($extensionPaths)) {
+            throw SnippetException::invalidExtensions($options->extensionPaths);
+        }
+
+        return $extensionPaths;
+    }
+
+    /**
+     * @param array{
+     *     domain?: string,
+     *     locale: string,
+     *     language: string,
+     *     script: ?string,
+     *     region: ?string,
+     *     isBase?: ?string,
+     * } $currentFileData
+     */
+    private function createTranslationFile(
```

Comment:
You could move this into the TranslationFile and use the create pattern instead. Also, I wouldn’t pass such a large array with lots of information that then gets validated or processed inside the function. It’s better to prepare the correct values beforehand and only pass those into the function. In my opinion, a method should always expect properly prepared values.